tests/migrations/helpers/test_update_legacy_filtered_positives_helper.py

```diff
@@ -0,0 +1,91 @@
+import pandas as pd
+import numpy as np
+from migrations.helpers.update_legacy_filtered_positives_helper import (
+    v0_version_set,
+    legacy_mongo_samples,
+    get_cherrypicked_samples_by_date,
+    split_mongo_samples_by_version,
+)
+from crawler.constants import (
+    FIELD_PLATE_BARCODE,
+    FIELD_ROOT_SAMPLE_ID,
+    V0_V1_CUTOFF_TIMESTAMP,
+    V1_V2_CUTOFF_TIMESTAMP,
+)
+
+# ----- migration helper function tests -----
+
+
+def test_legacy_mongo_samples_returns_correct_samples_filtered_by_date(
+    config, filtered_positive_testing_samples
+):  # noqa: E501
+    result = legacy_mongo_samples(config)
+    assert len(result) == 3
+
+
+def test_check_versions_set_returns_true_with_v0(config, filtered_positive_testing_samples):
+    assert v0_version_set(config) is True
+
+
+def test_check_versions_set_returns_false_with_no_v0_samples(
+    config, filtered_positive_testing_samples_no_v0
+):  # noqa: E501
+    assert v0_version_set(config) is False
+
+
+def test_get_cherrypicked_samples_by_date_v0_returns_expected(
+    config, event_wh_data, mlwh_sample_stock_resource, mlwh_sql_engine, event_wh_sql_engine
+):
+    root_sample_ids = ["root_1", "root_2", "root_3", "root_4"]
+    plate_barcodes = ["pb_1", "pb_2", "pb_3", "pb_4"]
+
+    expected_rows = [["root_1", "pb_1"], ["root_2", "pb_2"]]
+    expected_columns = [FIELD_ROOT_SAMPLE_ID, FIELD_PLATE_BARCODE]
+    expected = pd.DataFrame(np.array(expected_rows), columns=expected_columns, index=[0, 1])
+
+    returned_samples = get_cherrypicked_samples_by_date(
+        config, root_sample_ids, plate_barcodes, "1970-01-01 00:00:01", V0_V1_CUTOFF_TIMESTAMP
+    )
+    pd.testing.assert_frame_equal(expected, returned_samples)
+
+
+def test_get_cherrypicked_samples_by_date_v1_returns_expected(
+    config, event_wh_data, mlwh_sample_stock_resource, mlwh_sql_engine, event_wh_sql_engine
+):
+    root_sample_ids = ["root_1", "root_2", "root_3", "root_4"]
+    plate_barcodes = ["pb_1", "pb_2", "pb_3", "pb_4"]
+
+    expected_rows = [["root_1", "pb_1"], ["root_3", "pb_3"]]
```

Comment:
Nice spot - I've updated the query so that it looks like this:

f" AND mlwh_sample.created >= '{start_date}'"
f" AND mlwh_sample.created < '{end_date}'"

Anything created before and any time up to but not including the cut-off is defined as the version in question - does that make sense?

Reply:
Yep I think this makes sense. The upper bound is not inclusive, the lower bound is :+1: